src/datasource.js

```diff
@@ -176,7 +176,13 @@ export class HawkularDatasource {
       return flatTags.map(tag => {
         return {text: tag, value: tag};
       });
-    });
+    }));
+  }
+
+  runWithResolvedVariables(target, func) {
+    const resolved = this.variables.resolve(target, {});
+    return this.q.all(resolved.map(p => func(p)))
+      .then(result => [].concat.apply([], result));
```

Comment:
Oh I see, good hack, however it looks like there might be problems with big arrays but we can use [flattenDeep](https://lodash.com/docs/4.17.4#flattenDeep) (or just [flatten](https://lodash.com/docs/4.17.4#flatten)) from lodash.
```javascript
import {flattenDeep} from 'lodash';
...
return this.q.all(resolved.map(p => func(p)))
  then(result => flattenDeep(result));
```